homeassistant/components/pvpc_hourly_pricing/sensor.py

```diff
@@ -139,13 +142,12 @@ def __init__(
         self.entity_description = description
         self._attr_attribution = coordinator.api.attribution
         self._attr_unique_id = unique_id
-        self._attr_name = name
         self._attr_device_info = DeviceInfo(
             configuration_url="https://api.esios.ree.es",
             entry_type=DeviceEntryType.SERVICE,
             identifiers={(DOMAIN, coordinator.entry_id)},
             manufacturer="REE",
-            name="ESIOS API",
+            name=name,
```

Comment:
Hi @joostlek,

I'm not sure about this change, setting the variable 'name' (intended for the main price sensor) into the service  that _groups_ (potentially) multiple sensors. 

Right now there is only one (the 'PVPC' price sensor), but **there are more sensors to come** (in #85767 to implement authenticated access to API, and #85769 to add more price sensors, following a long-time feature request to show the injection price for solar installations).

In that scenario, as I understand (but I'm probably outdated 😓), the _service-type device_ should have a common name, to group all sensors, and each sensor should get its default name from the entity description key, am I wrong?

Reply:
Hey 👋🏻.

So the intention of this PR is to enable has_entity_name. Currently the name of this sensor is coming from the config entry data, which we actually want to stop doing, since if you really want to give a device a different prefix, you could rename the config entry or the device.

As I don't use PVPC and I am just on a quest to transform all integrations to has_entity_name = True, I want to know how you see this change fit best. For some more context about this change, checkout https://developers.home-assistant.io/docs/core/entity#has_entity_name-true-mandatory-for-new-integrations

Reply:
Thanks for the link and context, @joostlek  👌

> Currently the name of this sensor is coming from the config entry data, which we actually want to stop doing

In this integration, the `entry.data[CONF_NAME]` could be dropped anytime, as it's only used to have a default name for the sensor, which is also set in the `SensorEntityDescription.key`, and it's not translatable (it's an abr). 

In the proposed PRs to increase the number of sensors, I think it was changed to use just the description.key, but now I see the `has_entity_name` is also a field for the `SensorEntityDescription`, so I suppose it could be used like this:

```python
SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
    SensorEntityDescription(
        key="PVPC",
        icon="mdi:currency-eur",
        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
        state_class=SensorStateClass.MEASUREMENT,
        has_entity_name=True,
        name="PVPC",
    ),
)
```

and leave the device name with its current fixed name (or changing it to just 'ESIOS'):

```python
    self._attr_device_info = DeviceInfo(
            configuration_url="https://api.esios.ree.es",
            entry_type=DeviceEntryType.SERVICE,
            identifiers={(DOMAIN, coordinator.entry_id)},
            manufacturer="REE",
            name="ESIOS API",
        )
```

That way, the default sensor name will be renamed to "ESIOS API PVPC", and other future sensors will be named the same way 👍. 

Maybe cutting 'ESIOS API' to just 'ESIOS' 👌

And the config entry 'name' field will be ignored completely when naming sensors 👍

I'll try to adapt to these changes in the other PRs, to check how this entity naming works with the new sensors, please give me a couple of days to provide better feedback, if the proposal above is not good enough 🥺😜

Reply:
I'd suggest moving the has entity name to the class and when implementing the rest directly adapt to this. What does PVPC actually mean? After this PR I wanted to add entity translations so maybe you can use the full name as the translation value?

Reply:
> What does PVPC actually mean?

> maybe you can use the full name as the translation value?

It means 'precio voluntario para el pequeño consumidor' 🙈 (:= 'voluntary price for small consumers', for contracts with power < 15kW), and it's an _official denomination_ for the _official price_ of electricity that any distributor company in Spain has to offer, in addition to any 'commercial' price that vendors could also offer; so kind of an **official reference price**, used in a big % of houses in Spain.

But I can asure you that **nobody would know what it means** 😅, so IMHO it would be an error to expand it and add translations for it 😅

Reply:
hehe, I can imagine. The more you know.

But what about device name to ESIOS and then PVPC as name?

Reply:
> But what about device name to ESIOS and then PVPC as name?

I think that could do it 👍